plugins/chrome-ux-report-backend/src/service/Query.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Config } from '@backstage/config';
+import { BigQuery } from '@google-cloud/bigquery'; // used Google's official BigQuery SDK.
+
+function createBigQueryClient(config: Config) {
+  const projectId = config.getString('chromeUXReport.projectId');
+  const keyPath = config.getString('chromeUXReport.keyPath');
+
+  return new BigQuery({
+    projectId: projectId,
+    keyFilename: keyPath,
```

Comment:
does the BQ client support other auth variants? for customers that run their services on Google Cloud proper, they can often rely on automatic machine identity based auth (like, using service accounts, and supplied via the metadata service) so no additional work to provide auth is needed, isn't that right?